### PR TITLE
fix: return proper error if OIDC Discoverydoc fails to respond

### PR DESCRIPTION
**Pull request by jiuker**
## Description

return err with status

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression [issue17212](https://github.com/minio/minio/issues/17212)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: docs/sts/web-identity.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func parseDiscoveryDoc(ustr string) (DiscoveryDoc, error) {
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return d, err
+		return d, errors.New(resp.Status)
```
Comment (shtripat) [inline docs/sts/web-identity.go]: Suggestion, this way actual error is lost. Should we create final error with response and error both before return?

Comment (jiuker) [reply docs/sts/web-identity.go]: here err always nil @shtripat

File: internal/config/identity/openid/jwt.go [hunk 2]
```diff
@@ -279,7 +279,7 @@ func parseDiscoveryDoc(u *xnet.URL, transport http.RoundTripper, closeRespFn fun
 	}
 	defer closeRespFn(resp.Body)
 	if resp.StatusCode != http.StatusOK {
-		return d, err
+		return d, errors.New(resp.Status)
```
Comment (harshavardhana) [inline internal/config/identity/openid/jwt.go]: We should provide more context here by saying some words

```suggestion
		return d, fmt.Errorf("unexpected error returned by %s : status(%s)", u, resp.Status)  
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Review by shtripat — APPROVED** [review]
LGTM

**Pull request merged**